x-pack/legacy/plugins/infra/public/components/metrics_explorer/chart.tsx

```diff
@@ -101,7 +97,10 @@ export const MetricsExplorerChart = injectI18n(
                 id={getAxisId('timestamp')}
                 position={Position.Bottom}
                 showOverlappingTicks={true}
-                tickFormat={dateFormatter}
+                tickFormat={useCallback(
```

Comment:
Conditionally calling a hook violates [one of the invariants of the React hook system](https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level):

![image](https://user-images.githubusercontent.com/973741/59914101-f2328080-9419-11e9-8d98-fd4430d34d07.png)

It must remain outside of the conditional. Instead, we can move the conditional inside the hook:

```typescript
const dateFormatter = useMemo(
  () =>
    series.rows.length > 0
      ? niceTimeFormatter([first(series.rows).timestamp, last(series.rows).timestamp])
      : (value: number) => `${value}`,
  [series.rows]
);
```

Reply:
Doah! Good catch